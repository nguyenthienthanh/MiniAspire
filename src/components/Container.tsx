import React, { FC } from 'react'
import {
  StyleSheet,
  StatusBar,
  SafeAreaView,
  StatusBarProps,
  View,
} from 'react-native'
import { Colors } from 'react-native/Libraries/NewAppScreen'

export type ContainerProps = {
  barStyle?: StatusBarProps['barStyle']
}

const Container: FC<ContainerProps> = (props) => {
  return (
    <View style={styles.root}>
      <StatusBar barStyle={props.barStyle} translucent />
      <SafeAreaView>{props.children}</SafeAreaView>
    </View>
  )
}

Container.defaultProps = {
  barStyle: 'dark-content',
}

export default Container

const styles = StyleSheet.create({
  root: {
    backgroundColor: Colors.lighter,
  },
})
