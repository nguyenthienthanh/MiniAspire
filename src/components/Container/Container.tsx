import React, { FC } from 'react'
import {
  StyleSheet,
  StatusBar,
  SafeAreaView,
  StatusBarProps,
  View,
} from 'react-native'
import { useTheme } from '@react-navigation/native'

export type ContainerProps = {
  barStyle?: StatusBarProps['barStyle']
}

const Container: FC<ContainerProps> = (props) => {
  const { colors } = useTheme()

  return (
    <View style={{ backgroundColor: colors.background }}>
      <StatusBar
        barStyle={props.barStyle}
        translucent
        backgroundColor="transparent"
      />
      <SafeAreaView style={styles.container}>{props.children}</SafeAreaView>
    </View>
  )
}

Container.defaultProps = {
  barStyle: 'dark-content',
}

export default Container

const styles = StyleSheet.create({
  container: {
    paddingTop: StatusBar.currentHeight,
  },
})
