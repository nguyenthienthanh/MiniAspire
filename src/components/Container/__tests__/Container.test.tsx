/**
 * @format
 */

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

import Container from '../Container'

describe('Container', () => {
  it('should render correctly and match the snapshot', () => {
    const container = Enzyme.shallow(<Container />)
    expect(toJson(container)).toMatchSnapshot()
  })

  it('should render with status bar style is "light-content" correctly', () => {
    expect.assertions(1)

    const container = renderer.create(<Container barStyle="light-content" />)

    expect(container.root.findByProps({}).props.barStyle).toBe('light-content')
  })
})
