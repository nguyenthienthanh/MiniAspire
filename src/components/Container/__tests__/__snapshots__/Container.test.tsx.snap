// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render correctly and match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
    }
  }
>
  <StatusBar
    animated={false}
    backgroundColor="transparent"
    barStyle="dark-content"
    showHideTransition="fade"
    translucent={true}
  />
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "paddingTop": null,
      }
    }
  />
</View>
`;
